### DISCUSSION TITLE
Cursor Pagination and Pivot Ordering

### DISCUSSION THREAD

**Discussion by haleyngonadi** [laravel/framework#49714] [Q&A]
I have a belongsToMany relationship: 

```
public function shows()
    {
        return $this->belongsToMany(Show::class, 'widget_shows')->withPivot('order_column')->orderByPivot('order_column');
    }
```

I'm trying to paginate the results from that relationship like so:

```
$widget = Widget::findOrFail($id);

return $widget->shows()->cursorPaginate(10);
```

But it fails immediately, with the error:

> Unable to find parameter [widget_shows.order_column] in pagination item.

Am I missing something?

**Comment by macropay-solutions** [comment]
Try 

     return $this->belongsToMany(
            Show::class, // 4
            (new WidgetShows())->getTable() . ' as t', // 2 & 3
            't.?', // 2
            't.?', // 3
            '?', // 1 $this
            '?' // 4
        )->withPivot('t.order_column')->orderByPivot('t.order_column');
